code/iaas/service-discovery/server/src/main/java/io/cattle/platform/servicediscovery/process/RegionMonitor.java

```diff
@@ -105,24 +134,56 @@ public void run() {
                 boolean notFound = (externalProject == null && externalProjectResponse.getStatusCode()==200);
                 if(notFound) {
                         objectProcessManager.executeStandardProcess(StandardProcess.REMOVE, link, null);
+                        continue;
                 } else {
                     boolean uuidsDoNotMatch = (externalProject!=null &&
                             storedUUID!=null && !storedUUID.equals(externalProject.getUuid()));
                     if(uuidsDoNotMatch) {
                             objectProcessManager.executeStandardProcess(StandardProcess.REMOVE, link, null);
+                            continue;
                     }
                 }
+                if(link.getExternal()) {
+                        Account localAccount = objectManager.loadResource(Account.class, link.getAccountId());
+                        ExternalAccountLink accLink = RegionUtil.getAccountLinkForExternal(targetRegion, externalProject, localAccount, jsonMapper);
+                        if(accLink == null || removedStates.contains(accLink.getState())){
+                            objectProcessManager.executeStandardProcess(StandardProcess.REMOVE, link, null);
```

Comment:
lets put all links to remove into separate map instead of removing it immediately. Then go over the list and remove (try/catch individual remove)

Reply:
Just saw there's also 3-4 RegionUtil calls which require try and catch, hence the big try/catch. Or do we want to split em into each separate call - for eg - `checkRegion`